### PR TITLE
Update k8s.io/utils dependency version to latest

### PR DESCRIPTION
**Pull request by jingxu97**
This PR updates the utils vendor package to latest version.
run the following commands
./hack/pin-pin-dependency.sh k8s.io/utils 278ece378a5054f16a07622b51ddaf82b328d6e6
./hack/update-vendor.sh

The changes included in the updated utils include
1. mount utils: remove letter assignment during volume formatAndMount for windows.
2. mount utils: remove readlink check during isNotLilklyMountpoint for windows.

The latest commit for utils repo https://github.com/kubernetes/utils/commit/278ece378a5054f16a07622b51ddaf82b328d6e6

**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind deprecation
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

**Comment by jingxu97** [comment]
cc @dims

**Comment by jingxu97** [comment]
@dims @msau42 Could you help take a look?

**Comment by liggitt** [comment]
vendor mechanics look good

/cc @andyzhangx 
is there a windows presubmit we can run to exercise the modified mount code?

**Comment by andyzhangx** [comment]
> vendor mechanics look good
> 
> /cc @andyzhangx
> is there a windows presubmit we can run to exercise the modified mount code?

pls always try `/test pull-kubernetes-e2e-azure-disk-windows` when there is windows related change. Thanks.

**Comment by liggitt** [comment]
Can the job be configured to run on relevant patterns automatically, like `(volume|mount).*(azure|windows)`?

**Comment by andyzhangx** [comment]
I found an interesting issue in azure disk windows e2e test, with this PR, it costs 8m30s to 
 - delete a pod
 - create a new pod

While without this PR, it only costs 2m30s:

 - with this PR(8m30s):
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/91468/pull-kubernetes-e2e-azure-file-windows/1266547927606628352/build-log.txt
```
May 30 12:07:14.887: INFO: Deleting pod "azuredisk-volume-tester-qtppf-778f5dd694-smjbt" in namespace "azuredisk-2167"
May 30 12:07:14.946: INFO: Waiting for pod "azuredisk-volume-tester-qtppf-778f5dd694-smjbt" in namespace "azuredisk-2167" to be fully deleted
STEP: checking again that the pod is running
STEP: checking pod exec
May 30 12:15:49.277: INFO: Running '/usr/local/bin/kubectl --kubeconfig=/root/tmp701128827/kubeconfig/kubeconfig.westus2.json exec azuredisk-volume-tester-qtppf-778f5dd694-fl4m2 --namespace=azuredisk-2167 -- powershell.exe -Command Get-Content C:\mnt\test-1\data.txt'
```

 - without this PR(2m30s):
https://prow.k8s.io/log?id=1266719180309139461&job=pull-kubernetes-e2e-azure-disk-windows
```
May 30 14:35:06.694: INFO: Deleting pod "azuredisk-volume-tester-72sbn-65dd88cdfc-44994" in namespace "azuredisk-1150"
May 30 14:35:06.726: INFO: Waiting for pod "azuredisk-volume-tester-72sbn-65dd88cdfc-44994" in namespace "azuredisk-1150" to be fully deleted
STEP: checking again that the pod is running
STEP: checking pod exec
May 30 14:37:36.891: INFO: Running '/usr/local/bin/kubectl --kubeconfig=/root/tmp085428885/kubeconfig/kubeconfig.southcentralus.json exec azuredisk-volume-tester-72sbn-65dd88cdfc-9bv5d --namespace=azuredisk-1150 -- powershell.exe -Command Get-Content C:\mnt\test-1\data.txt'
```

**Comment by jingxu97** [comment]
pull-kubernetes-e2e-windows-gce

**Comment by andyzhangx** [comment]
I got the error why it took so long time:
```
Events:
  Type     Reason       Age                 From                 Message
  ----     ------       ----                ----                 -------
  Normal   Scheduled    31s                 default-scheduler    Successfully assigned azuredisk-5859/azuredisk-volume-tester-khsgk-85d7978c66-k72pk to 1520k8s000
  Warning  FailedMount  11s (x11 over 30s)  kubelet, 1520k8s000  MountVolume.MountDevice failed for volume "pvc-b0a38912-fc7f-4173-90d7-6bef5f268a11" : azureDisk - mountDevice:IsLikelyNotMountPoint failed with readlink error: readlink \var\lib\kubelet\plugins\kubernetes.io\azure-disk\mounts\m2503046119: The system cannot find the path specified.
```

**Comment by andyzhangx** [comment]
with this PR, unmount would fail since `UnmountDevice` --> `doCleanupMountPoint` --> `mounter.IsLikelyNotMountPoint` and `mounter.IsLikelyNotMountPoint` would return error, I think aws, gce disk would have same issue since we use same implementation on `UnmountDevice`:
https://github.com/kubernetes/kubernetes/blob/46d08c89ab9f55bcaf23f0aa3742c53a72a7418a/pkg/volume/azure_dd/attacher.go#L287-L295

```
W0531 04:40:34.315721    4464 attacher.go:292] azureDisk - Device \var\lib\kubelet\plugins\kubernetes.io\azure-disk\mounts\m337994334 failed to unmount with error: readlink error: readlink \var\lib\kubelet\plugins\kubernetes.io\azure-disk\mounts\m337994334: The system cannot find the path specified.
```

 - all azure disk UnmountDevice failed with this PR:
```
azureuser@1520k8s000 C:\var\lib\kubelet\plugins\kubernetes.io\azure-disk\mounts>dir
 Volume in drive C is Windows
 Volume Serial Number is CA78-414E

 Directory of C:\var\lib\kubelet\plugins\kubernetes.io\azure-disk\mounts

05/31/2020  11:23 AM    <DIR>          .
05/31/2020  11:23 AM    <DIR>          ..
05/31/2020  11:03 AM    <SYMLINKD>     m1303142941 [\\?\Volume{dd90e517-0000-0000-0000-100000000000}\]
05/31/2020  05:38 AM    <SYMLINKD>     m1540271893 [\\?\Volume{1d718a80-0000-0000-0000-100000000000}\]
05/31/2020  11:23 AM    <SYMLINKD>     m2339616850 [\\?\Volume{c53807e4-0000-0000-0000-100000000000}\]
05/31/2020  11:10 AM    <SYMLINKD>     m2365177410 [\\?\Volume{ca6cb186-0000-0000-0000-100000000000}\]
05/31/2020  05:04 AM    <SYMLINKD>     m2432319735 [\\?\Volume{0b0d1c5a-0000-0000-0000-100000000000}\]
05/31/2020  04:43 AM    <SYMLINKD>     m2503046119 [\\?\Volume{caaaba86-0000-0000-0000-100000000000}\]
05/31/2020  04:56 AM    <SYMLINKD>     m2588452738 [\\?\Volume{7648ac6c-0000-0000-0000-100000000000}\]
05/31/2020  04:40 AM    <SYMLINKD>     m337994334 [\\?\Volume{edcc53eb-0000-0000-0000-100000000000}\]
05/31/2020  05:17 AM    <SYMLINKD>     m3757038432 [\\?\Volume{5b550651-0000-0000-0000-100000000000}\]

```

File: vendor/k8s.io/utils/mount/mount_windows.go
Comment (andyzhangx) [inline vendor/k8s.io/utils/mount/mount_windows.go:243]: this will cause Unmount failure, since original `Unmount` will remove `driveLetter + ":"` simply, now the target is volumeId
```
// Unmount unmounts the target.
func (mounter *Mounter) Unmount(target string) error {
	klog.V(4).Infof("azureMount: Unmount target (%q)", target)
	target = NormalizeWindowsPath(target)
	if output, err := exec.Command("cmd", "/c", "rmdir", target).CombinedOutput(); err != nil {
		klog.Errorf("rmdir failed: %v, output: %q", err, string(output))
		return err
	}
	return nil
}
```

Comment (jingxu97) [reply vendor/k8s.io/utils/mount/mount_windows.go:243]: so with volumeId, unmount will fail? 
I think I manually tested it and it can pass. I will test it again.

**Comment by andyzhangx** [comment]
This is really weird, I am trying to debug this issue, although `\var\lib\kubelet\plugins\kubernetes.io\azure-disk\mounts\m3940783135` is a valid link from host, while `mountDevice:IsLikelyNotMountPoint` invoked by kubelet thought it's invalid, see below:
```
Events:
  Type     Reason       Age               From                 Message
  ----     ------       ----              ----                 -------
  Normal   Scheduled    <unknown>         default-scheduler    Successfully assigned default/busybox-azuredisk-0 to 1520k8s000
  Warning  FailedMount  4s (x8 over 67s)  kubelet, 1520k8s000  MountVolume.MountDevice failed for volume "pvc-907376dd-165d-4b00-872e-5146eb0b4221" : azureDisk - mountDevice:IsLikelyNotMountPoint failed with readlink error: readlink \var\lib\kubelet\plugins\kubernetes.io\azure-disk\mounts\m3940783135: The system cannot find the path specified.

  
 azureuser@1520k8s000 C:\Users\azureuser>cd \var\lib\kubelet\plugins\kubernetes.io\azure-disk\mounts\m3940783135
azureuser@1520k8s000 C:\var\lib\kubelet\plugins\kubernetes.io\azure-disk\mounts\m3940783135>dir
 Volume in drive C is Windows
 Volume Serial Number is CA78-414E

 Directory of C:\var\lib\kubelet\plugins\kubernetes.io\azure-disk\mounts\m3940783135

06/01/2020  02:52 AM                 4 abc
               1 File(s)              4 bytes
               0 Dir(s)  107,268,366,336 bytes free
```

- another issue:
I found the `mklink` command sometimes failed while it does not report any error:
https://github.com/kubernetes/kubernetes/blob/6fb37b85190442cd6260571ae3496eb4ee7ecc45/vendor/k8s.io/utils/mount/mount_windows.go#L245-L247

in below error, `bind` failed since above `mklink` command actually failed while it did not report any error.
```
 Warning  Failed                  0s (x2 over 1s)  kubelet, 1520k8s000      Error: Error response from daemon: invalid volume specification: 'c:\var\lib\kubelet\pods\7a381965-575e-4bbf-a7bd-cb9a3895b34b\volumes\kubernetes.io~azure-disk\pvc-959dd867-91dc-4a55-b675-2c21d43b4c15:c:/mnt/azuredisk': invalid mount config for type "bind": bind source path does not exist: c:\var\lib\kubelet\pods\7a381965-575e-4bbf-a7bd-cb9a3895b34b\volumes\kubernetes.io~azure-disk\pvc-959dd867-91dc-4a55-b675-2c21d43b4c15
```

I had a feeling that `mklink` command with volumeid it not stable, it sometimes has weird issue.

File: vendor/k8s.io/utils/mount/mount_windows.go
Comment (andyzhangx) [reply vendor/k8s.io/utils/mount/mount_windows.go:243]: I am pretty sure the unmount would fail with this PR, I have reproed the unmount failure again in my env. `IsLikelyNotMountPoint` func does not work well with volumeID, here is the error:
```
azureDisk - TearTownAt:IsLikelyNotMountPoint check failed: CreateFile c:\\var\\lib\\kubelet\\pods\\84444347-81ef-48de-84d9-702f43295565\\volumes\\kubernetes.io~azure-disk\\pvc-2f4ae459-3cc0-419b-a1e8-5b0a29acfa30: The system cannot find the file specified."
```

**Comment by andyzhangx** [comment]
here is the fix about the unmount volumeid failure issue: https://github.com/kubernetes/utils/pull/169,

**Comment by andyzhangx** [comment]
with the fix(https://github.com/kubernetes/utils/pull/169), that case in `pull-kubernetes-e2e-azure-disk-windows` could pass very quickly:

```
STEP: deleting the pod for deployment
Jun  3 07:28:23.457: INFO: Deleting pod "azuredisk-volume-tester-4777c-598d6768fb-zxf68" in namespace "azuredisk-987"
Jun  3 07:28:23.462: INFO: Waiting for pod "azuredisk-volume-tester-4777c-598d6768fb-zxf68" in namespace "azuredisk-987" to be fully deleted
STEP: checking again that the pod is running
STEP: checking pod exec
Jun  3 07:28:35.478: INFO: Running '/usr/local/bin/kubectl --kubeconfig=/root/.kube/config exec azuredisk-volume-tester-4777c-598d6768fb-xdhc6 --namespace=azuredisk-987 -- powershell.exe -Command Get-Content C:\mnt\test-1\data.txt'
```

**Comment by jpbetz** [comment]
https://storage.googleapis.com/kubernetes-jenkins/pr-logs/pull/91468/pull-kubernetes-e2e-windows-gce/1268277003442720768/build-log.txt

/retest

**Comment by thockin** [comment]
Would be nice if the PR message said WHY we are updating this

/approve

**Comment by andyzhangx** [comment]
@jingxu97 the `mount_windows.go` is not updated with https://github.com/kubernetes/utils/pull/169, would you update this PR?

**Comment by jpbetz** [comment]
We're waiting on this for https://github.com/kubernetes/kubernetes/pull/88936 as well. Can we get this updated to https://github.com/kubernetes/utils/pull/169 and get it passing?

/priority important-soon
/kind cleanup

**Comment by jingxu97** [comment]
pull-kubernetes-e2e-azure-file-windows

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Comment by andyzhangx** [comment]
@jingxu97 are you going to fix this `pull-kubernetes-e2e-windows-gce` test failure? or merge this PR first?

**Comment by BenTheElder** [comment]
https://github.com/kubernetes/kubernetes/pull/92063

**Comment by jingxu97** [comment]
> @jingxu97 are you going to fix this `pull-kubernetes-e2e-windows-gce` test failure? or merge this PR first?
> This PR would break the subPath on disk?

I think it will break subPath tests. We will try to discuss possible solutions in csi and storage meetings this week.

**Comment by msau42** [comment]
I think there are other folks waiting on this update to get changes unrelated to mount utils. To unblock them, can we revert the mount updates and then get the mount changes into the new mount-utils staging area that we're creating in https://github.com/kubernetes/kubernetes/pull/91596? This is exactly one of the reasons why we want to get mount out of k8s/utils and into its own repo.

**Comment by jpbetz** [comment]
Any update on this?

**Comment by jingxu97** [comment]
I have a PR to roll back utils https://github.com/kubernetes/utils/pull/173

**Comment by jpbetz** [comment]
@jingyxu97 I'll bump utils to the lastest (with the breaking change reverted) via https://github.com/kubernetes/kubernetes/pull/91634.

Should this be closed?

**Comment by cheftako** [comment]
This seems to be a bit of a smattering of fixes. We seem to have a window mount fix, trace changes, utils dependency changes and some typo fixes. It would probably be good to separate as many of these fixes as possible. Smaller changes are easier to review. Its easier to deal with conflicts if there are fewer changes. You will need few approvals, so its easier to get it merged. Your also not conflating unrelated changes, so its less likely any given (sub) change will be revoked.

**Pull request closed without merging**
